server/superdesk/io/commands/remove_expired_content.py

```diff
@@ -58,31 +62,36 @@ def remove_expired_data(provider):
             if not item.get('archived'):
                 for file_id in [rend.get('media') for rend in item.get('renditions', {}).values()
                                 if rend.get('media')]:
+                    print('Deleting file: ', file_id)
                     superdesk.app.media.delete(file_id)
 
     stats.incr('ingest.expired_items', items.count())
-    print('Removed expired content for provider: %s' % provider['_id'])
+    print('Removed expired content for provider: %s' % provider.get('_id', 'Detached items'))
 
 
 def get_expired_items(provider_id, expiration_date):
     query_filter = get_query_for_expired_items(provider_id, expiration_date)
     req = ParsedRequest()
-    req.max_results = 100
+    req.max_results = 25
     req.args = {'filter': query_filter}
     return superdesk.get_resource_service('ingest').get(req, None)
 
 
-def get_query_for_expired_items(provider_id, expiration_date):
+def get_query_for_expired_items(provider, expiration_date):
     """Find all ingest items with given provider id and
     (expiry is past or
     (no expiry assigned and versioncreated is less then calculated expiry date))"""
-    query = {'and': [
-        {'term': {'ingest.ingest_provider': provider_id}},
-        {'or': [
-            {'range': {'ingest.expiry': {'lte': date_to_str(utcnow())}}},
-            {'and': [{'missing': {'field': 'ingest.expiry'}},
-                     {'range': {'ingest.versioncreated': {'lte': date_to_str(expiration_date)}}}]},
-        ]},
+    query = {'should': [
+        {'range': {'ingest.expiry': {'lte': date_to_str(utcnow())}}},
+        {'and': [{'missing': {'field': 'ingest.expiry'}},
+                 {'range': {'ingest.versioncreated': {'lte': date_to_str(expiration_date)}}}]},
     ]}
 
-    return superdesk.json.dumps(query)
+    if provider.get('_id'):
+        query['must'] = {'term': {'ingest.ingest_provider': str(provider.get('_id'))}}
+
+    if provider.get('exclude'):
```

Comment:
This "exclude" section is needed to find items with unknown providers. Is that right?